### ISSUE TITLE
HttpClient timeout handling

### ISSUE THREAD

**Issue by ayende** [dotnet/corefx#36005]
Currently, `HttpClient` has the `Timeout` property, which reflects the overall time that a single request can take.
This result in an issue when using large requests / response. Because the value is the overall time for the request, we have to set it to a high value.
However, there are other impacts here. If we suffer server based packet loss, the high timeout will also mean that we are never getting any replies back. 

You can test that disconnecting the _server_ from the network and observing that the request is still in flight.

I tried to use `ResponseDrainTimeout`, but that doesn't do things properly (https://gist.github.com/ayende/7fbbb32219ee1d4860bb8f2e8b9d724b).
Basically, the issue is that I want to be able to set (separately):

* `Socket.RecieveTiemout`
* `Socket.SendTiemout`
* `Timeout` - for the entire request 

There used to be `ReadWriteTimeout` which would have worked, but it is gone in `SocketsHttpHandler` and is ignored in `HttpWebRequest` in Core CLR.

This has a major effect on the robustness of the code in the face of bad network conditions.

**Comment by stephentoub** [comment]
Dup of https://github.com/dotnet/corefx/issues/33845?

**Comment by ayende** [comment]
Yes, I believe so

**Comment by stephentoub** [comment]
Thanks.  Closing as a dup.
